File: restore.js
const fs = require('fs');

const logPath = 'C:\\Users\\aou\\.gemini\\antigravity\\brain\\973eca96-7195-45cc-99ac-babef94fe98d\\.system_generated\\logs\\transcript.jsonl';

const lines = fs.readFileSync(logPath, 'utf8').split('\n');
let diffOutput = null;

for (const line of lines) {
    if (line.includes('@@ -630,984 +630,6 @@')) {
        try {
            const data = JSON.parse(line);
            function findDiff(d) {
                if (typeof d === 'object' && d !== null) {
                    if (d.output && typeof d.output === 'string' && d.output.includes('@@ -630,984 +630,6 @@')) {
                        return d.output;
                    }
                    for (const key in d) {
                        const res = findDiff(d[key]);
                        if (res) return res;
                    }
                } else if (Array.isArray(d)) {
                    for (const v of d) {
                        const res = findDiff(v);
                        if (res) return res;
                    }
                }
                return null;
            }
            diffOutput = findDiff(data);
            if (diffOutput) break;
        } catch (e) {
            // ignore JSON parse errors
        }
    }
}

if (diffOutput) {
    const diffLines = diffOutput.split('\n');
    let extracted = [];
    let inBlock = false;
    for (const line of diffLines) {
        if (line.startsWith('@@ -630,984 +630,6 @@')) {
            inBlock = true;
            continue;
        }
        if (inBlock) {
            if (line.startsWith('@@')) break;
            if (line.startsWith('-')) {
                extracted.push(line.substring(1));
            }
        }
    }
    fs.writeFileSync('recovered.js', extracted.join('\n'));
    console.log(`Recovered ${extracted.length} lines.`);
} else {
    console.log("Could not find the diff output.");
}

File: scripts/restore.py
import json

log_path = r'C:\Users\aou\.gemini\antigravity\brain\973eca96-7195-45cc-99ac-babef94fe98d\.system_generated\logs\transcript.jsonl'

diff_output = ""
with open(log_path, 'r', encoding='utf-8') as f:
    for line in f:
        if '@@ -630,984 +630,6 @@' in line:
            data = json.loads(line)
            # Find the output string in the json structure
            def find_diff(d):
                if isinstance(d, dict):
                    if 'output' in d and isinstance(d['output'], str) and '@@ -630,984 +630,6 @@' in d['output']:
                        return d['output']
                    for k, v in d.items():
                        res = find_diff(v)
                        if res: return res
                elif isinstance(d, list):
                    for v in d:
                        res = find_diff(v)
                        if res: return res
                return None
            diff_output = find_diff(data)
            if diff_output:
                break

if diff_output:
    lines = diff_output.split('\n')
    extracted = []
    in_block = False
    for line in lines:
        if line.startswith('@@ -630,984 +630,6 @@'):
            in_block = True
            continue
        if in_block:
            if line.startswith('@@'):
                break
            # If line is deleted, we want to keep it
            if line.startswith('-'):
                extracted.append(line[1:]) # remove the '-'
            elif line.startswith(' '):
                # Context line, do not include in our extracted block because they weren't deleted
                pass

    with open('recovered.txt', 'w', encoding='utf-8') as f:
        f.write('\n'.join(extracted))
    print(f"Recovered {len(extracted)} lines into recovered.txt")
else:
    print("Could not find the diff output.")
